models/UserManager.php

```diff
@@ -0,0 +1,289 @@
+    }
+
+    // Fonction qui vérifie qu'un utilisateur est bien modérateur ou admin
+    function is_modo($email,$token){
+        $tableau = [
+            'email' =>  $email,
+            'token' =>  $token
+        ];
+        $sql = "SELECT * FROM admins WHERE email=:email AND token=:token";
+        $req= $this->db->prepare($sql);
+        $req->execute($tableau);
+        
+        return $req->rowCount($sql);
+    }
+
+    // Fonction permettant la mise à jour du mot de passe 
+    function update_password($password){
+         $tableau = [
+            'password'  =>  sha1($password),
+            'session'   =>  $_SESSION['admin']
+        ];
+
+        $sql = "UPDATE admins SET password = :password WHERE email=:session";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+
+    }
+
+    // Fonction permettant de récupérer la liste des administrateurs
+    function get_user(){
+        $req = $this->db->query("
+            SELECT * FROM admins WHERE email='{$_SESSION['admin']}';
+        ");
+
+        $result = $req->fetchObject();
+        
+        return $result;
+    }
+    
+    function login_verification(){
+       if(filter_has_var(INPUT_POST, 'submit')){
+            if(filter_has_var(INPUT_POST, 'email')){
+                $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email', FILTER_VALIDATE_EMAIL)));
+            }
+            if(filter_has_var(INPUT_POST, 'password')){
+                $password = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password')));
+            }
+            
+            $errors = [];
+
+            if(empty($email) || empty($password)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis!";
+            }else if(Usermanager::is_admin($email,$password) == 0){
+                $errors['exist']  = "Cet administrateur n'existe pas";
+            }
+                    
+            if(empty($errors)){
+                $_SESSION['admin'] = $email;
+                header("Location:/dashboard");
+            }
+            return $errors;
+        }
+    }
+    
+    function settings_verification(){
+          if(filter_has_var(INPUT_POST, 'submit')){
+                if(filter_has_var(INPUT_POST, 'name')){
+                    $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name')));
+                }
+                if(filter_has_var(INPUT_POST, 'email')){
+                    $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+                }
+                if(filter_has_var(INPUT_POST, 'email_again')){
+                    $email_again = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email_again', FILTER_VALIDATE_EMAIL)));
+                }
+                if(filter_has_var(INPUT_POST, 'role')){
+                    $role = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'role')));
+                }
+                
+                $token = UserManager::token(30);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class 'UserManager' in method 'settings_verification'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)